### PR TITLE
CATROID-249 Colorpicker option for show variable

### PR DESCRIPTION
**Pull request by ThomasSchwengler**
* Add colorpicker dependency to all flavors
* Add a custom dialog style for dialogs with title, mirroring the one in pocket paint
* Add tests for basic colorpicker functionality
* Show a selection alertdialog, asking the user if they want to pick a new value in the colorpicker or edit it in the formula editor

### CODE REVIEW

File: catroid/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1979,6 +1979,7 @@ needs read and write access to it. You can always change permissions through you
     <string name="brick_show_variable_aligned_centered">centered</string>
     <string name="brick_show_variable_aligned_left">left</string>
     <string name="brick_show_variable_aligned_right">right</string>
+    <string name="brick_context_dialog_pick_color">Pick Color</string>
```
Comment (wslany) [inline catroid/src/main/res/values/strings.xml]:
```suggestion
    <string name="brick_context_dialog_pick_color">Pick color</string>
```
Please use standard English capitalization rules at all places. 

Note: If you find places where we have missed it, please correct it. Also in Paintroid please.

Comment (ThomasSchwengler) [reply catroid/src/main/res/values/strings.xml]: Fixed

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/ShowTextColorSizeAlignmentBrick.java [hunk 2]
```diff
@@ -83,14 +70,85 @@ public ShowTextColorSizeAlignmentBrick(int xPosition, int yPosition, double size
...
+	private void switchSelectEditDialogOption(View view, int which) {
+		if (which == 0) {
+			showColorPicker(view);
+		} else if (which == 1) {
+			showFormulaEditor(view);
+		}
+	}
+
+	private void showColorPicker(View view) {
+		AppCompatActivity activity = UiUtils.getActivityFromView(view);
```
Comment (84n4n4) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/ShowTextColorSizeAlignmentBrick.java]: please nullcheck

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/ShowTextColorSizeAlignmentBrick.java [hunk 3]
```diff
@@ -83,14 +70,85 @@ public ShowTextColorSizeAlignmentBrick(int xPosition, int yPosition, double size
...
+	private int getCurrentColor() {
+		String formulaString = getColorFormulaString();
+		boolean validColorString = formulaString != null && formulaString.length() == 7 && formulaString.matches("^#[0-9a-fA-F]+$");
+		return validColorString ? Color.parseColor(formulaString) : Color.BLACK;
+	}
+
+	private @Nullable String getColorFormulaString() {
+		Formula formula = getFormulaWithBrickField(BrickField.COLOR);
+		FormulaElement formulaTree = formula.getFormulaTree();
+		boolean formulaIsString = formulaTree.getElementType() == FormulaElement.ElementType.STRING;
```
Comment (84n4n4) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/ShowTextColorSizeAlignmentBrick.java]: please replace with if else construct

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**